src/modules/commander/commander.cpp

```diff
@@ -2533,6 +2535,21 @@ int commander_thread_main(int argc, char *argv[])
 			}
 		}
 
+		/* reset mission state when disarmed */
```

Comment:
Why is this logic in the commander after all? Can't the navigator just do this by itself, since it knows the current flight mode?